docs/ldml/tr35.md

```diff
@@ -2972,6 +3036,43 @@ The binary operators '&', '-', and the implicit union have equal precedence and
 
 **One caution:** the '&' and '-' operators operate between sets. That is, they must be immediately preceded and immediately followed by a set. For example, the pattern **[[:Lu:]-A]** is illegal, since it is interpreted as the set **[:Lu:]** followed by the incomplete range **-A**. To specify the set of upper case letters except for 'A', enclose the 'A' in brackets: **[[:Lu:]-[A]]**.
 
+##### <a name="Variables_in_UnicodeSets" href="#Variables_in_UnicodeSets">Variables in UnicodeSets</a>
+
+Support for variable identifiers (var) is optional.
+They are used in certain contexts such as in [https://cldr-smoke.unicode.org/spec/main/ldml/tr35-general.html#Transforms](Transforms).
+When they are used, they are defined as follows:
+
+UnicodeSets may contain variables (`$my_char`, `$the_set`, ...) in place of full UnicodeSets and strings/characters.
+
+Implementations that support variables must require the value of variables to be known at UnicodeSet parse time.
+Parsing a UnicodeSet with variables support requires referenced variables to exist, in other words, parsing a UnicodeSet referring to an unknown variable leads to a parse error.
+
+If the values of variables were unknown at parse time, this would introduce ambiguities when parsing, thus the values must be known.
+For example, it would be unclear whether **[$a-$b]** is a range of strings/characters, or a set difference, because that depends on the values of the variables.
+In particular in the case of **$a-$b**, it must be that **$a** and **$b** are of the same kind, i.e., both are strings/characters or both are UnicodeSets. A variable map with any other combination (e.g., `{ a => '0', b => [:L:] }`) is incompatible with the syntax **$a-$b**, and would return an error at parse-time.
```

Comment:
Reading this again it feels a bit odd, because usually we don’t expose a separate parsing operation, only « turn this string in UnicodeSet syntax into an actual set of characters and strings », which obviously requires the values of the variables.

How about this, trimming the discussion of the reduce-reduce conflict and keeping it a note, and putting the emphasis on the fact that variable assignments can be invalid, which is what really matters to the user, as opposed to what might matter to the implementer:
```suggestion
Not all variable maps are valid for a given expression in UnicodeSet syntax.
For instance, consider **[$a-$b]**; this may be a range of characters if both **$a** and **$b** are characters,
or a difference of sets if they are both sets; but given the map `{ a => '0', b => [:L:] }`, it is invalid.

> **Note:** In particular, the variable map is needed not just to compute the actual set of characters and strings represented by the UnicodeSet,
but also to parse the UnicodeSet syntax: if **$a** and **$b** are unknown, the parsing of **[$a-$b]** is ambiguous.
```

Reply:
Thanks! Tried something in https://github.com/unicode-org/cldr/pull/3089/commits/8f504903fc089f4b72d67c3f5acd6662bfad57f8 :)